pokemon-checker/src/DataTransferObjects/AbilityDTO.ts

```diff
@@ -0,0 +1,37 @@
+export type ListOfPokemon = {
+  name: string;
+  url: string;
+}
+
+export type AbilityConstructorOptions = {
+  name: string,
+  url: string,
+  id?: number,
+  effect?: string,
+  pokemons?: ListOfPokemon[],
+}
+
+export class AbilityDTO {
+  public name: string;
+  public url: string;
+  public id: number;
+  public effect: string;
+  public pokemons: ListOfPokemon[];
+  public isHidden: boolean;
```

Comment:
Should we add a public property to indicate whether an ability DTO is a full one, or a stub?

Just so anything handling the data knows whether we need to fetch the effect + other info to display it.

Reply:
Starting a review so we don't forget to make this change.

An example of this is written in the `PokemonDTO` under `hasFullData()`.

Reply:
I like the idea, although its been a while and I can't remember the reasoning for stubbing before full creation, is it to catch missing data from the actual object?

Reply:
Exactly that. When we get a pokemon object back, it'll usually have just a name+url for the ability.

If we want to include ability effects or hidden status, then the view needs to know whether that data is missing or not - it doesn't come for free when we fetch a pokemon, so we need to fetch the full ability.